File: specification/devtestlabs/resource-manager/Microsoft.DevTestLab/stable/2018-09-15/examples-js/VirtualNetworks_Update.js
const { DevTestLabsClient } = require("@azure/arm-devtestlabs");
const { DefaultAzureCredential } = require("@azure/identity");

/**
 * This sample demonstrates how to Allows modifying tags of virtual networks. All other properties will be ignored.
 *
 * @summary Allows modifying tags of virtual networks. All other properties will be ignored.
 * x-ms-original-file: specification/devtestlabs/resource-manager/Microsoft.DevTestLab/stable/2018-09-15/examples/VirtualNetworks_Update.json
 */
async function virtualNetworksUpdate() {
  const subscriptionId = "{subscriptionId}";
  const resourceGroupName = "resourceGroupName";
  const labName = "{labName}";
  const name = "{virtualNetworkName}";
  const virtualNetwork = {
    tags: { tagName1: "tagValue1" },
  };
  const credential = new DefaultAzureCredential();
  const client = new DevTestLabsClient(credential, subscriptionId);
  const result = await client.virtualNetworks.update(
    resourceGroupName,
    labName,
    name,
    virtualNetwork
  );
  console.log(result);
}

virtualNetworksUpdate().catch(console.error);

File: specification/devtestlabs/resource-manager/Microsoft.DevTestLab/stable/2018-09-15/examples-java/VirtualNetworks_Update.java

import com.azure.resourcemanager.devtestlabs.models.VirtualNetwork;
import java.util.HashMap;
import java.util.Map;

/**
 * Samples for VirtualNetworks Update.
 */
public final class Main {
    /*
     * x-ms-original-file:
     * specification/devtestlabs/resource-manager/Microsoft.DevTestLab/stable/2018-09-15/examples/VirtualNetworks_Update
     * .json
     */
    /**
     * Sample code: VirtualNetworks_Update.
     * 
     * @param manager Entry point to DevTestLabsManager.
     */
    public static void virtualNetworksUpdate(com.azure.resourcemanager.devtestlabs.DevTestLabsManager manager) {
        VirtualNetwork resource = manager.virtualNetworks().getWithResponse("resourceGroupName", "{labName}",
            "{virtualNetworkName}", null, com.azure.core.util.Context.NONE).getValue();
        resource.update().withTags(mapOf("tagName1", "tagValue1")).apply();
    }

    // Use "Map.of" if available
    @SuppressWarnings("unchecked")
    private static <T> Map<String, T> mapOf(Object... inputs) {
        Map<String, T> map = new HashMap<>();
        for (int i = 0; i < inputs.length; i += 2) {
            String key = (String) inputs[i];
            T value = (T) inputs[i + 1];
            map.put(key, value);
        }
        return map;
    }
}
